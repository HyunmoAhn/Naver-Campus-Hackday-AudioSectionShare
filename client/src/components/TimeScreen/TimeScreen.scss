.TimeScreen {
  color: #B0B0B0;
  width: 45px;
  padding: 5px;
  margin: 5px;
  -ms-user-select: none;
  -moz-user-select: -moz-none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  user-select: none;

  font-family: "Courier", "Monaco", "Consolas", "Courier New";
}