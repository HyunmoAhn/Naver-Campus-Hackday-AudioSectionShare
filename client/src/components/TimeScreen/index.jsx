import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './TimeScreen.scss';

const propTypes = {
	className: PropTypes.string,
	second: PropTypes.number,
};

const defaultProps = {
	className: '',
	second: 0,
};

class TimeScreen extends React.Component {
	render() {
		const { className, second } = this.props;
		const number = parseInt(second);
		const minute = number / 60;
		const remainSecond = number % 60;
		const resultSecond = remainSecond >= 10 ? remainSecond : `0${remainSecond}`;

		if (isNaN(second)) {
			return null;
		}

		return (
			<span className={cx('TimeScreen', className)}>
				{parseInt(minute)}:{resultSecond}
			</span>
		)
	}
}

TimeScreen.propTypes = propTypes;
TimeScreen.defaultProps = defaultProps;

export default TimeScreen;