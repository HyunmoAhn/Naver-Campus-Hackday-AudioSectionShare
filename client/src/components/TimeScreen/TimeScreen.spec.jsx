import React from 'react';
import { shallow } from 'enzyme';

import TimeScreen from './';

describe('<TimeScreen />', () => {
	it('should match snapshot when render default', () => {
		const wrapper = shallow(<TimeScreen />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match snapshot when props.second is NaN', () => {
		const wrapper = shallow(<TimeScreen second={0 / 0} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match snapshot when remainSecond >= 10', () => {
		const wrapper = shallow(<TimeScreen second={35} />);

		expect(wrapper).toMatchSnapshot();
	});
});