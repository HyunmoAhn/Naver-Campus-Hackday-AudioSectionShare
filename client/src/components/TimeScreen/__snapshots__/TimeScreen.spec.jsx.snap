// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeScreen /> should match snapshot when props.second is NaN 1`] = `""`;

exports[`<TimeScreen /> should match snapshot when remainSecond >= 10 1`] = `
<span
  className="TimeScreen"
>
  0
  :
  35
</span>
`;

exports[`<TimeScreen /> should match snapshot when render default 1`] = `
<span
  className="TimeScreen"
>
  0
  :
  00
</span>
`;
